docs/spec/callables.rst

```diff
@@ -347,60 +403,355 @@ generate errors in such cases.
+        ...
 
-  def async_query(on_success: Callable[[int], None],
-                  on_error: Callable[[int, Exception], None]) -> None:
-      # Body
+    x: Callable[[], str]
 
-It is possible to declare the return type of a callable without
-specifying the call signature by substituting a literal ellipsis
-(three dots) for the list of arguments::
+Parameters specified using ``Callable`` are assumed to be positional-only.
+The ``Callable`` form provides no way to specify keyword-only parameters,
+variadic parameters, or default argument values. For these use cases, see
+the section on `Callback protocols`_.
 
-  def partial(func: Callable[..., str], *args) -> Callable[..., str]:
-      # Body
+Meaning of ``...`` in ``Callable``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Note that there are no square brackets around the ellipsis.  The
-arguments of the callback are completely unconstrained in this case
-(and keyword arguments are acceptable).
+The ``Callable`` special form supports the use of ``...`` in place of the
+list of parameter types. This indicates that the type is consistent with
+any input signature::
 
-Since using callbacks with keyword arguments is not perceived as a
-common use case, there is currently no support for specifying keyword
-arguments with ``Callable``.  Similarly, ``Callable`` does not support
-specifying callback signatures with a variable number of arguments of a
-specific type. For these use cases, see the section on
-`Callback protocols`_.
+    cb1: Callable[..., str]
+    cb1 = lambda x: str(x)  # OK
+    cb1 = lambda : ""  # OK
+
+    cb2: Callable[[], str] = cb1  # OK
+
+The input signature ``(...)`` is not the same as ``(*args: Any, **kwargs: Any)``.
+The former is a gradual type form that is consistent with all input signatures,
+and all input signatures are consistent with it. The latter is not a gradual
+type form. It is consistent with all input signatures, but the converse is
+not true. The relationship between ``(...)`` and ``(*args: Any, **kwargs: Any)``
+is similar to the relationship between ``Any`` and ``object``. Just as ``Any``
+means "any conceivable type that could be compatible", ``(...)`` means "any
+conceivable set of parameters that could be compatible".
+
+The ``...`` syntax can also be used to provide a :ref:`specialized value for a
+ParamSpec <paramspec_valid_use_locations>` in a generic class or type alias.
+For example::
+
+    type Callback[**P] = Callable[P, str]
+
+    def func(cb: Callable[[], str]) -> None:
+        f: Callback[...] = cb  # OK
+
+If ``...`` is used with signature concatenation, the ``...`` portion continues
+to mean "any conceivable set of parameters that could be compatible"::
+
+    type CallbackWithInt[**P] = Callable[Concatenate[int, P], str]
+    type CallbackWithStr[**P] = Callable[Concatenate[str, P], str]
+
+    def func(cb: Callable[[int, str], str]) -> None:
+        f1: CallbackWithInt[...] = cb  # OK
+        f2: CallbackWithStr[...] = cb  # Error
```

Comment:
I think you should also provide a simple example which just directly uses `Concatenate` without an intermediary `ParamSpec`. I had to read this example a couple of times, to convince myself that these examples are sufficient to illustrate that `Concatenate[int, ...]` also behaves in this way.
```suggestion
If ``...`` is used with signature concatenation, the ``...`` portion continues
to mean "any conceivable set of parameters that could be compatible"::

    type CallbackWithInt[**P] = Callable[Concatenate[int, P], str]
    type CallbackWithStr[**P] = Callable[Concatenate[str, P], str]

    def func(cb: Callable[[int, str], str]) -> None:
        f1: Callable[Concatenate[int, ...], str] = cb # OK
        f2: Callable[Concatenate[str, ...], str] = cb # Error
        f3: CallbackWithInt[...] = cb  # OK
        f4: CallbackWithStr[...] = cb  # Error
```